experimental/Schemes/src/MaxContact.jl

```diff
@@ -0,0 +1,318 @@
+###################################################################
+## Tools for maximal contact objects                              
+## - getters for Scheme, Covering, current ambient space etc
+## - constructors  and updating 
+###################################################################
+
+###################################################################
+## Getters
+###################################################################
+original_scheme(MCO::Oscar.MaxContactObject) = MCO.W_orig
+covering(MCO::Oscar.MaxContactObject) = MCO.C
+maximal_contact_data(MCO::Oscar.MaxContactObject) = MCO.max_contact_data
+
+ambient_generators(MCC::Oscar.MaxContactChart) = MCC.ambient_gens
+ambient_orders(MCC::Oscar.MaxContactChart) = MCC.ambient_orders
+dependent_variables(MCC::Oscar.MaxContactChart) = MCC.dependent_vars
+max_contact_minor_data(MCC::Oscar.MaxContactChart) = MCC.minor_data
+prepared_jacobi_matrices(MCC::Oscar.MaxContactChart) = MCC.ambient_jacobi
+
+function current_hypersurface_sequences(MC::Oscar.MaxContactObject)
+  max_contact_data = maximal_contact_data(MC)
+  res_dict = IdDict{AbsAffineScheme,Vector{MPolyRingElem}}()
+  for U in keys(max_contact_data)
+    res_dict[U] = ambient_generators(max_contact_data[U])
+  end
+  return res_dict
+end
+
+function current_order_sequences(MC::Oscar.MaxContactObject)
+  max_contact_data = maximal_contact_data(MC)
+  res_dict = IdDict{AbsAffineScheme, Vector{Int}}()
+  for U in keys(max_contact_data)
+    res_dict[U] = ambient_orders(max_contact_data[U])
+  end
+  return res_dict
+end
+
+###################################################################
+## Constructors and Updating
+###################################################################
+function _initialize_max_contact_object(inc::Oscar.CoveredClosedEmbedding)
+
+  W = codomain(inc)
+
+  ## the empty hulls for the Oscar.MaxContactObject --  to be filled in
+  max_contact_data = IdDict{AbsAffineScheme,Oscar.MaxContactChart}()
+  patch_list = AbsAffineScheme[]
+
+  ## run through the charts and set up a complete intersection covering
+  ## for the locally complete intersection
+  for U in affine_charts(W)
+    IU = modulus(OO(U))
+
+    if ngens(IU) == 0 
+    ## trivial case -- nothing to be done
+      max_contact_data[U] = Oscar.MaxContactChart(U,[],[],[],[],[])
+      continue
+    end
+
+    ## find minors of jacobian matrix involved in expressing 1
+    RU = base_ring(OO(U))
+    IUgens = lifted_numerator.(gens(IU))
+    JM = jacobian_matrix(IUgens)
+    min_list = [a for a in minors_with_position(JM, nvars(RU) - dim(OO(U))) if !is_zero(a[1])]
+    min_id = ideal(RU,[a[1] for a in min_list])
+    min_id_full = min_id+ideal(RU,IUgens)+modulus(OO(U))
+    radical_membership(one(RU),min_id_full)  || error("image of embedding not smooth")
+    coord_vec = coordinates(one(RU),min_id_full)
+    nonzero_indices =  [k for k in 1:ngens(min_id) if !is_zero(coord_vec[k])]
```

Comment:
We have not streamlined this yet over all types of ideals, but I suppose that we encounter only polynomial ideals here. Hence `coord_vec` should be an `SRow` and we can do 
```suggestion
    coord_vec = coordinates(one(RU),min_id_full)
    nonzero_indices =  [k for (k, _) in coord_vec if k <= ngens(min_id)]
```